src/layers/Layer.cpp

```diff
@@ -0,0 +1,19 @@
+#include "layers/Layer.hpp"
+
+size_t Shape::get_index(const std::vector<size_t>& coords) const {
```

Comment:
if you refactor FCLayer, please move `mat_vec_mul` function to FCLayer too

Reply:
Ok